### PR TITLE
feat: User notification endpoint returns settings for all projects

### PR DESCRIPTION
**Pull request by lynnagara**
This PR modifies the user notification endpoint to return the full set of user notification settings. It now includes all of the per-project notification settings in addition to the global user defaults

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/api/endpoints/user_notification_details.py [hunk 1]
```diff
@@ -55,20 +63,44 @@ def get_attrs(self, item_list, user, *args, **kwargs):
...
+            return bool(int(val))  # '1' is true, '0' is false
+        elif (external_type == int):
+            return int(val)
+        else:
+            return val
+
     def serialize(self, obj, attrs, user, *args, **kwargs):
 
-        raw_data = {option.key: option.value for option in attrs}
+        data = defaultdict(lambda: {})
```
Comment (MaxBittker) [inline src/sentry/api/endpoints/user_notification_details.py]: i think you can use defaultdict(dict)

**Review by MaxBittker — COMMENTED** [review]
what is a high level description of what this PR does?

File: src/sentry/api/endpoints/user_notification_details.py (hunk 1 above)
Comment (mattrobenolt) [reply src/sentry/api/endpoints/user_notification_details.py]: Yes.

Comment (lynnagara) [reply src/sentry/api/endpoints/user_notification_details.py]: 😀

**Comment by billyvg** [comment]
Going to close this in favor of a new endpoint to handle per project/org 
 options: https://github.com/getsentry/sentry/pull/7240/files#diff-e99482e539a95210c0717c02c42055adR115

**Pull request closed without merging**